src/utils/processOutputInterpreters/trivy-image-output.ts

```diff
@@ -0,0 +1,112 @@
+const LineSplitter = /\r?\n/;
+const dateHeader = /^[-\d]+T[-:.\d]+Z?\s+\033\[\d+m([A-Z]+)\033\[\d+m\s+(.*)$/;
+const borderPattern = /^\+(?:-+\+){6}\s*$/;
+const internalBorderPattern = /^\+(?:(?:-+|\s+)\+){6}\s*$/;
+const rowPattern = /^(?:\|[^|]+){6}\|\s*$/;
+
+const CVEKeys = ['Package', 'VulnerabilityID', 'Severity', 'Title', 'InstalledVersion', 'FixedVersion', 'Description', 'PrimaryURL'];
+const severityRatings: Record<string, number> = {
+  LOW:      1,
+  MEDIUM:   2,
+  HIGH:     3,
+  CRITICAL: 4,
+  UNKNOWN:  5,
+};
+const MaxSeverityRating = Math.max(...Object.values(severityRatings));
+
+type finalVulType = Record<string, string>;
+
+export default class TrivyScanImageOutputCuller {
+  prelimLines: Array<string>;
+  JSONLines: Array<string>;
+  inJSON = false;
+
+  constructor() {
+    this.prelimLines = [];
+    this.JSONLines = [];
+  }
+
+  getRating(key: string) {
+    return key in severityRatings ? severityRatings[key] : MaxSeverityRating;
+  }
+
+  fixLines(lines: Array<string>) {
+    // "key": "value with an escaped \' single quote isn't valid json"
+    return lines.map(line => line.replace(/\\'/g, "'"));
+  }
+
+  addData(data: string): void {
+    if (this.inJSON) {
+      this.JSONLines.push(data.replace(/\\'/g, "'"));
+
+      return;
+    }
+    const lines = data.split(LineSplitter);
+    const jsonStartIndex = lines.indexOf('[');
+
+    if (jsonStartIndex >= 0) {
+      this.prelimLines = this.prelimLines.concat(lines.slice(0, jsonStartIndex));
+      this.inJSON = true;
+      this.JSONLines = this.fixLines(lines.slice(jsonStartIndex));
+    } else {
+      this.prelimLines = this.prelimLines.concat(lines);
+    }
+  }
+
+  getProcessedData() {
+    const prelimLines = this.prelimLines.map(line => line.replace(dateHeader, '$1 $2'));
```

Comment:
So this drops the timestamp, keeps the severity (as in `INFO`, `WARN`, etc.), and the rest of the message right?  Calling it `dateHeader` was a bit misleading; maybe `logFormat`?

Reply:
True...